streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/python/PythonGateway.java

```diff
@@ -0,0 +1,143 @@
+package org.ray.streaming.runtime.python;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.msgpack.core.Preconditions;
+import org.ray.api.annotation.RayRemote;
+import org.ray.streaming.api.context.StreamingContext;
+import org.ray.streaming.python.PythonFunction;
+import org.ray.streaming.python.PythonPartition;
+import org.ray.streaming.python.stream.PythonStreamSource;
+import org.ray.streaming.util.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gateway for streaming python api.
+ * All calls on DataStream in python will be mapped to DataStream call in java by this
+ * PythonGateway using ray calls.
+ * <p>
+ * Note: this class needs to be in sync with `ray.streaming.gateway_client.GatewayClient`
+ */
+@SuppressWarnings("unchecked")
+@RayRemote
+public class PythonGateway {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PythonGateway.class);
+  private static final String REFERENCE_ID_PREFIX = "__gateway_reference_id__";
+
+  private MsgPackSerializer serializer;
+  private Map<String, Object> referenceMap;
+  private StreamingContext streamingContext;
+
+  public PythonGateway() {
+    serializer = new MsgPackSerializer();
+    referenceMap = new HashMap<>();
+    LOGGER.info("PythonGateway created");
+  }
+
+  public byte[] createStreamingContext() {
+    streamingContext = StreamingContext.buildContext();
+    LOGGER.info("StreamingContext created");
+    referenceMap.put(getReferenceId(streamingContext), streamingContext);
+    return serializer.serialize(getReferenceId(streamingContext));
+  }
+
+  public byte[] withConfig(byte[] confBytes) {
+    Preconditions.checkNotNull(streamingContext);
+    try {
+      Map<String, String> config = (Map<String, String>) serializer.deserialize(confBytes);
+      LOGGER.info("Set config {}", config);
+      streamingContext.withConfig(config);
+      return new byte[1];
```

Comment:
why?

Reply:
It's a limitation in ray actor call, java ray call doesn't support `return void` and `return new byte[0]` will crash at `ray::CoreWorker::ExecuteTask`.  So I work around by using `return new byte[1].`  I'll add some comments.